src/Symfony/Component/Validator/Constraints/NoSuspiciousCharactersValidator.php

```diff
@@ -70,12 +74,7 @@ public function validate(mixed $value, Constraint $constraint)
         if ($constraint->restrictionLevel) {
             $checker->setRestrictionLevel($constraint->restrictionLevel);
         }
-
-        $allowedLocales = $constraint->profileLocales;
-        if ($constraint->addCurrentLocaleToProfile) {
-            $allowedLocales[] = $this->locale;
-        }
-        $checker->setAllowedLocales(implode(',', $allowedLocales));
+        $checker->setAllowedLocales(implode(',', $constraint->profileLocales ?? $this->enabledLocales));
```

Comment:
Just "locales" in both cases?

Reply:
If empty, skip the call?

Reply:
> Just "locales" in both cases?

Regarding `profileLocales` I think it’s better to introduce the concept of [profile](https://www.unicode.org/reports/tr39/#Identifier_Characters) that way. The term is used in `CHECK_CHAR_LIMIT` phpDoc and it does not depend solely on locale ([ICU allows to define one using characters](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4j/com/ibm/icu/text/SpoofChecker.Builder.html#setAllowedChars-com.ibm.icu.text.UnicodeSet-) e.g.)

`enabledLocales`  could be renamed to `defaultProfileLocales` to explicit how and when it’s used :thinking: 

> If empty, skip the call?

This would not make any functional difference so I don’t know :man_shrugging:

Reply:
To me, the profile is the full set of attributes on the constraint. Repeating the word for one random property doesn't look DX friendly IMHO.
`#[NoSuspiciousCharacters(locales: ['en', 'fr'])]` looks sweet.

OK for `$defaultLocales` for the service.

> This would not make any functional difference so I don’t know man_shrugging

WDYT? Calling setAllowedLocales with the empty array already accepts any locales? If yes, then fine ofc.

Reply:
> To me, the profile is the full set of attributes on the constraint.

Indeed, I misunderstood what a profile was. :expressionless: 

> Calling setAllowedLocales with the empty array already accepts any locales?

Yes I checked and this is the same behavior described [here](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4j/com/ibm/icu/text/SpoofChecker.Builder.html#setAllowedLocales-java.util.Set-):

> Supplying an empty string removes all restrictions; characters from any script will be allowed.